src/tests/Interop/PInvoke/Int128/Int128Test.cs

```diff
@@ -25,6 +32,47 @@ unsafe partial class Int128Native
     [DllImport(nameof(Int128Native))]
     public static extern Int128 AddInt128(Int128 lhs, Int128 rhs);
 
+
+    [DllImport(nameof(Int128Native))]
+    public static extern void AddStructWithInt128_ByRef(ref StructWithInt128 lhs, ref StructWithInt128 rhs);
```

Comment:
Should pointers also be checked to make sure the marshaler doesn't change the by-ref marshalling?

Reply:
I've done some mild testing of pointers. It should work acceptably well.